src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_jni.h

```diff
@@ -521,3 +522,23 @@ jmethodID GetOptionalMethod(JNIEnv *env, bool isStatic, jclass klass, const char
 jfieldID GetField(JNIEnv *env, bool isStatic, jclass klass, const char* name, const char* sig);
 JNIEnv* GetJNIEnv(void);
 int GetEnumAsInt(JNIEnv *env, jobject enumObj);
+
+static inline void
+do_assert_always (bool condition, const char* fmt, ...)
+{
+    if (condition) {
+        return;
+    }
+
+    va_list ap;
+
+    va_start (ap, fmt);
+    __android_log_vprint (ANDROID_LOG_FATAL, "DOTNET", fmt, ap);
+    va_end (ap);
+
+    abort ();
+}
+
+#define assert_always(_condition_, _fmt_, ...) do_assert_always (_condition_, "%s:%d (%s): " _fmt_, __FILE__, __LINE__, __FUNCTION__, ## __VA_ARGS__)
+#define assert_always_valid_pointer_argument(_ptr_) assert_always ((_ptr_) != NULL, "Parameter '%s' must be a valid pointer", #_ptr_)
+#define assert_always_positive_integer_argument(_arg_) assert_always ((_arg_) > 0, "Parameter '%s' must be larger than 0", #_arg_)
```

Comment:
Just so you know, we also have assert definitions for various platforms defined here https://github.com/dotnet/runtime/blob/3d8de060ea419fafd82fb4039aafb3afdbfb938f/src/libraries/Native/Unix/Common/pal_compiler.h#L11-L19 (i.e. accessible via `#include "pal_compiler.h"`, in case you want to reorganize similar macro definitions together for readability, code discovery etc.)

Reply:
yeah I think we can move it to pal_compiler.h

what do you think about using `fail_if` instead of `assert_always`? that reads better to me:
```
fail_if()
fail_if_invalid_pointer_argument()
fail_if_negative_integer_argument()
```

Reply:
yup, sounds good

Reply:
Just pushed the change and realized it's a bad one :( `abort_if` (or `failed_if`) suggests that the function will abort/fail if the condition is **true**, which is not was intended. If we want to use this syntax, we should use `fail/abort_unless`

Reply:
Unless we reverse the condition...

Reply:
Ah, ok, I wasn't sure that's why I checked for the standard. I'll fix it up in a sec

Reply:
hm, it appears lack of the condition breaks the build:

```
/__w/1/s/src/libraries/Native/Unix/Common/pal_compiler.h:6:10: fatal error: 'stdbool.h' file not found
  #include <stdbool.h>
```